input/pagecontent/shorthand.md

```diff
@@ -0,0 +1,924 @@
+* us-core-ethnicity.detailed[0].valueCoding = RACE#2184-0 "Dominican"
+* us-core-ethnicity.detailed[1].valueCoding = RACE#2148-5 "Mexican"
+* us-core-ethnicity.text.valueString = "Hispanic or Latino"
+* us-core-birthsex.valueCode = #F
+```
+
+#### Structure Definition Escape Paths
+
+FSH uses the carat (^) syntax to provide direct access to any element in StructureDefinition. The carat syntax is the method for setting metadata attributes in SD (attributes not associated with any element). The carat syntax can be combined with non-carat (element) paths to set values in the SD associated with a particular element (see example below).
+
+**Example** Setting metadata attributes in a profile:
+
+```
+Profile:        USCorePatient
+Parent:         Patient
+* ^description = "Defines constraints and extensions on the patient resource for the minimal set of data to query and retrieve patient demographic information."
+* ^id = "us-core-patient"
+* ^status = #active
+* ^experimental = false
+```
+
+**Example** Setting metadata attribute for value set:
+```
+ValueSet:  BirthSex
+* ^id = "birthsex"
+GEN#F   "Female"
+GEN#M   "Male"
+NUL#UNK "Unknown"
+```
+
+**Example** Addressing SD attributes in an individual element definition:
+```
+Profile:        USCorePatient
+Parent:         Patient
+* communication.language ^binding.extension[0].url = "http://hl7.org/fhir/StructureDefinition/elementdefinition-maxValueSet"
+```
+***
+## Rules
+
+Rules are the mechanism for constraining a profile, defining an extension, creating slices, and more. All rules begin with an asterisk:
+
+`* {rule statement}`
+
+Here is a summary of the rules supported in FSH:
+
+| Rule Type | Rule Syntax | Examples |
+| --- | --- |---|
+| Fixed value |`* {path} = {value}`  | `* experimental = true` <br/> `* status = #active` <br/> `* valueString = "foo"` <br/> `* valueQuantity.value = 1.23` |
+| Value set binding |`* {path} from {valueSet} ({strength})`| `* telecom.system from http://hl7.org/fhir/ValueSet/contact-point-system (required)`
+| Narrowing cardinality | `* {path} {min}..{max}` | `* identifier 1..*` <br/> `* identifier.system 1..1`
+| Data type restriction | `* {path} only {type1} or {type2} or {type3}` | `* value[x] only code` <br/> `* value[x] only code or Quantity` |
+| Reference type restriction | `* {path} only Reference({type1} | {type2} | {type3})` | `* subject only Reference(Patient)` <br/> `* basedOn only Reference(MedicationRequest | ServiceRequest)` |
+| Flag assignment | `* {path} {flag1} {flag2}` <br/> `* {path1}, {path2}, {path3}... {flag}` | `* communication MS ?!` <br> `* identifier, identifier.system, identifier.value, name, name.family MS`
+| Invariants | `* obeys {invariant}` <br/> `* {path} obeys {invariant}` <br/> `* {path1}, {path2}, ... obeys {invariant}` | `* name obeys USCoreNameInvariant` |
+| Slicing | `* {array element path} contains {sliceName1} {card1} and {sliceName2} {card2} ...` | `* component contains SystolicBP 1..1 and DiastolicBP 1..1` |
```

Comment:
This brings up the need for indication for required versus optional elements. Here is a set of conventions: http://docopt.org/.  @cmoesel do you think we should adopt these?  I'm a bit hesistant because we use square brackets in the language itself.

Reply:
Grahame doesn't seem to follow much of any convention. For example, describing the syntax of mapping language (https://www.hl7.org/fhir/mapping-language.html), i see things like this:

`context.element { : type {min..max}} {default [value]} { list-option } as variable where [FHIRPath] check [FHIRPath]`

with the following example:

` src.value : integer 0..* default 10 first as vs0   where value >= 10 check value <= 100  log value`

Certainly can't differentiate between something to substitute (src) and something literally part of the statement (. : or default)